host/CMakeLists.txt

```diff
@@ -145,12 +145,22 @@ add_dependencies(oehost posix_untrusted_edl)
 # the CMake package.
 if (UNIX)
   find_library(CRYPTO_LIB NAMES crypto)
-  add_library(crypto SHARED IMPORTED)
-  set_target_properties(crypto PROPERTIES IMPORTED_LOCATION ${CRYPTO_LIB})
-
+  if (NOT CRYPTO_LIB)
```

Comment:
This is great. Can you copy this into the CMake package logic too? https://github.com/microsoft/openenclave/blob/1922dfbfc37c2d78361e662de45cd313af12f486/cmake/openenclave-config.cmake.in#L16

Reply:
I can add the crypto and dl as dependecies too. From what I know the find_dependency macro also checks for errors, so there is no need to also add conditions here. ([find dependecy macro](https://cmake.org/pipermail/cmake-developers/2016-February/027876.html))

Reply:
`find_dependency` only wraps `find_package`, which you said there weren't any for dl or crypto, so I think we'd have to use `find_library` here like you did outside of the package...

Reply:
I understand. I did the changes to the ```openenclave-config.cmake.in```.